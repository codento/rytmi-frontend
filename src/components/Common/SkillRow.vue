<template>
  <b-row
    v-b-tooltip.hover.top="description"
    :disabled.sync="disableTooltip"
    class="my-1"
    align-v="center"
  >
    <b-col
      cols="4"
      class="text-right"
    >
      {{ skillName(skillId) }}
    </b-col>
    <b-col
      cols="8"
      @mouseenter="active = id"
      @mouseleave="active = null"
    >
      <b-progress
        :max="5"
        :class="showSkillsOnly ? 'my-2' : 'mt-1'"
      >
        <b-progress-bar
          :value="knows"
          variant="c-orange"
        >
          <slot
            v-if="active !== id"
            name="label"
          >
            {{ knows ? knows.toString() : '' }}
          </slot>
        </b-progress-bar>
        <span
          v-show="active === id"
          class="skilldesc-center"
        >
          {{ knowsToDesc.text }}
        </span>
      </b-progress>
      <b-progress
        v-show="!showSkillsOnly"
        :max="5"
        class="mb-1"
      >
        <b-progress-bar
          :value="wantsTo"
          variant="c-orange-light"
        >
          <slot
            v-if="active !== id"
            name="label"
          >
            {{ wantsTo ? wantsTo.toString() : '' }}
          </slot>
        </b-progress-bar>
        <span
          v-show="active === id"
          class="skilldesc-center"
        >{{ wantsToDesc.text }}</span>
      </b-progress>
    </b-col>
  </b-row>
</template>

<script>
import { mapGetters } from 'vuex'
import proficiencyDesc from '@/assets/proficiencyDesc'

export default {
  name: 'SkillRow',
  props: {
    id: Number,
    skillId: Number,
    wantsTo: Number,
    knows: Number,
    description: String,
    highlight: Boolean,
    showSkillsOnly: {
      type: Boolean,
      default: false
    },
    disableTooltip: {
      type: Boolean,
      default: false
    },
    selectable: {
      type: Boolean,
      default: false
    }
  },
  data () {
    return {
      active: null
    }
  },
  computed: {
    ...mapGetters([
      'skillById',
      'skillName'
    ]),
    knowsToDesc: function () {
      return proficiencyDesc.knows['en'][this.knows]
    },
    wantsToDesc: function () {
      return proficiencyDesc.wants[this.wantsTo]
    }
  },
  methods: {
    selectSkill (skillId) {
      this.$emit('skill-selected', skillId)
    }
  }
}
</script>

<style scoped>
.skilldesc-center {
  position: absolute;
  left: 0;
  right: 0;
  text-align: center;
}
</style>
