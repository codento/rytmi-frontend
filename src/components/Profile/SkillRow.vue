<template>
  <b-row class="" style="padding-bottom: 12px" v-b-tooltip.hover.top="desc">
    <b-col class="col-mb-4 text-right valign-middle" style="text-size: 20px;">
      <b>{{ name }}:</b>
    </b-col>
    <b-col class="col-mb-8" @mouseenter="active = skillId" @mouseleave="active = null">
      <b-progress :max="5" class="mb-6" height="0.9rem">
        <b-progress-bar class="rytmi-progress-knowledge"
          :value="knows">
          <slot name="label" v-if="active !== skillId">{{ knows ? knows.toString() : '' }}</slot>
        </b-progress-bar>
        <span v-if="active === skillId" class="skilldesc-center">{{ knowsToDesc }}</span>
      </b-progress>
      <b-progress :max="5" class="mb-6" height="0.9rem">
        <b-progress-bar class="rytmi-progress-wants"
          :value="wants">
          <slot name="label" v-if="active !== skillId">{{ wants ? wants.toString() : '' }}</slot>
        </b-progress-bar>
        <span v-if="active === skillId" class="skilldesc-center">{{ wantsToDesc }}</span>
      </b-progress>
    </b-col>
  </b-row>
</template>

<script>
import { mapGetters } from 'vuex'
import proficiencyDesc from '../../assets/proficiencyDesc'

export default {
  name: 'SkillRow',
  props: {
    'skillId': Number,
    'name': String,
    'wants': Number,
    'knows': Number,
    'desc': String
  },
  data () {
    return {
      active: null
    }
  },
  computed: {
    ...mapGetters([
      'skillById'
    ]),
    knowsToDesc: function () {
      return proficiencyDesc.knows[this.knows]
    },
    wantsToDesc: function () {
      return proficiencyDesc.wants[this.wants]
    }
  },
  methods: {
  }
}
</script>

<style>
.tooltip-inner {
    max-width: 550px;
    width: inherit;
}
</style>

<style scoped>
.rytmi-progress-knowledge {
  background-color: #fda708;
}
.rytmi-progress-wants {
  background-color: #fedfa7;
}
.valign-middle {
  margin-top: auto;
  margin-bottom: auto;
}
.skilldesc-center {
  position: absolute;
  left: 0;
  right: 0;
  text-align: center;
}
</style>
