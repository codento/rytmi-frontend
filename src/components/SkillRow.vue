<template>
  <b-row>
    <b-col class="col-mb-4" style="text-size: 20px;">
      <b>{{ name }}:</b>
    </b-col>
    <b-col class="col-mb-8" >
      <b-progress
        height="0.9rem"
        :value="proficiency*20"
        variant='primary'
        class="mb-6"
      />
    </b-col>
  </b-row>
</template>
<script>
import { mapGetters } from 'vuex'
export default {
  name: 'SkillRow',
  props: {
    'proficiency': Number,
    'name': String,
    'skillId': Number
  },
  computed: {
    ...mapGetters([
      'skillById'
    ])
  },
  created () {
    this.$store.dispatch('fetchSkills')
  }
}
</script>
<style scoped>

</style>
