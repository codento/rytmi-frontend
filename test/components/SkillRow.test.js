import {
  mount
} from 'vue-test-utils'
import SkillRow from '../../src/components/SkillRow.vue'

describe('UserProfileCard.test.js', () => {
  it('Template is correct', () => {
    let wrapper = mount(SkillRow, {
      propsData: {
        basicInformation: {
          proficiency: 121231312,
          name: 'mail'
        }
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
