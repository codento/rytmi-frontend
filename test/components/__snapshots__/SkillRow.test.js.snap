// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileCard.test.js Template is correct 1`] = `"<b-row style=\\"padding-bottom: 12px;\\"><b-col class=\\"col-mb-4\\"><b>:</b></b-col> <b-col class=\\"col-mb-8\\"><div><div><b-progress height=\\"0.9rem\\" value=\\"NaN\\" variant=\\"primary\\" class=\\"mb-6\\"></b-progress> <b-progress height=\\"0.9rem\\" value=\\"NaN\\" variant=\\"success\\" class=\\"mb-6\\"></b-progress></div></div></b-col></b-row>"`;
